resource/Resource.md

```diff
@@ -0,0 +1,154 @@
+# Resource API Overview
+The resource library primarily defines a type that captures information about the entity
+for which stats or traces are recorded. It further provides a framework for detection of
+resource information from the environment and progressive population as signals propagate
+from the core instrumentation library to a backend's exporter.
+
+## Resource type
+A `Resource` describes the entity for which a signal was collected through two fields:
+* `type`: a string which describes a well-known type of entity. It SHOULD be namespaced
+to avoid collisions across different environment, e.g. `k8s.io/container`,
+`cloud.google.com/gce/instance`. 
+* `labels`: a dictionary of labels with string keys and values that provide information
+about the entity.
+
+Type strings and label keys MAY start with a domain name and MAY be further namespaced through
+single slashes. All other characters MUST be alphanumeric. Label values MAY contain any valid
+unicode character.
+
+Implementations MAY define a `Resource` data type, constructed from the parameters above.
+`Resource` MUST have getters for retrieving all the information used in `Resource` definition.
+
+Example in Go:
+```go
+type Resource {
+	Type   string
+	Labels map[string]string
+}
+```
+
+## Populating resources
+Resource information MAY be populated at any point between startup of the instrumented
+application and passing it to a backend-specific exporter. This explicitly includes
+the path through future OpenCensus components such as agents or services.
```

Comment:
Does the property like "health" be a valid field for it? If one want to mark the node as unhealthy and then healthy again? Or it's set once, change never kind of properties?

Reply:
In general, volatile metadata isn't helpful in identifying a resource – so I'd say no. At the same time, such data could be useful for filtering noisy/spammy signals before ingesting them.

So I think it should be allowed but exporters wouldn't actually make this data part of signal data itself.
Basically, for the near future, only static properties are of relevance IMO.

Reply:
so resource may have property whether it's "master" or configuration "version" that is currently being used. 

Since this PR defines resources - I'd suggest to define resources lifetime and health as well.

Mostly my comment comes from the term "resource". It is so wide and general that I natirally see the desire to use it to design health and lifetime management of that resource.

Reply:
I think whether a resource has health or lifetime info to begin with really depends on the resource. If it has this information, a library implementing a resource detector can easily provide this info through dedicated labels.

Though, as hinted at in other comments, I think things should mostly be constrained to static resource information. Backends can pull into volatile metadata at query time from other sources.

Reply:
health and lifetime tracking requires more responsive mechanism to detect changes that pulling. If this is a scenario - it should be taken into consideration